dali/pipeline/operator/sequence_operator.h

```diff
@@ -0,0 +1,451 @@
+
+  bool Setup(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
+    expand_ = ShouldExpand(ws);
+    bool is_inferred;
+    if (!expand_) {
+      is_inferred = Operator<Backend>::Setup(output_desc, ws);
+    } else {
+      SetupSequenceOperator(ws);
+      SetupExpandedWorkspace(expanded_, ws);
+      ExpandInputs(ws);
+      ExpandArguments(ws);
+      is_inferred = Operator<Backend>::Setup(output_desc, expanded_);
+    }
+    return ProcessOutputDesc(output_desc, ws, is_inferred);
+  }
+
+  void Run(workspace_t<Backend> &ws) override {
+    if (!expand_) {
+      Operator<Backend>::Run(ws);
+    } else {
+      ExpandOutputs(ws);
+      Operator<Backend>::Run(expanded_);
+      PostprocessOutputs(ws);
+      Clear();
+    }
+  }
+
+  bool IsExpanded() const {
+    return expand_;
+  }
+
+  virtual bool ShouldExpandChannels() const {
+    return false;
+  }
+
+ protected:
+  virtual bool ShouldExpand(const workspace_t<Backend> &ws) {
+    auto num_inputs = ws.NumInput();
+    for (int input_idx = 0; input_idx < num_inputs; input_idx++) {
+      const auto &layout = GetInputLayout(ws, input_idx);
+      auto layout_desc =
+          ShouldExpandChannels() ? LayoutDesc::FrameAndChannel(layout) : LayoutDesc::Frame(layout);
+      if (layout_desc.NumDims() > 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  virtual void SetupSequenceOperator(const workspace_t<Backend> &ws) {
+    auto num_inputs = ws.NumInput();
+    input_expand_desc_.reserve(num_inputs);
+    for (int input_idx = 0; input_idx < num_inputs; input_idx++) {
+      const auto &input_shape = ws.GetInputShape(input_idx);
+      const auto &layout = GetInputLayout(ws, input_idx);
+      DALI_ENFORCE(layout.size() == 0 || layout.size() == input_shape.sample_dim(),
+                   make_string("Layout of input ", input_idx, " has size ", layout.size(),
+                               " which does not match the dimensionality of the input (got input "
+                               "with dimensionality ",
+                               input_shape.sample_dim(), ")."));
+      auto layout_desc =
+          ShouldExpandChannels() ? LayoutDesc::FrameAndChannel(layout) : LayoutDesc::Frame(layout);
+      input_expand_desc_.emplace_back(input_shape, layout_desc);
+    }
+  }
+
+  virtual void ExpandInputs(const workspace_t<Backend> &ws) {
+    for (int input_idx = 0; input_idx < ws.NumInput(); input_idx++) {
+      const auto &expand_desc = GetInputExpandDesc(input_idx);
+      auto num_expand_dims = expand_desc.NumDims();
+      if (ws.template InputIsType<GPUBackend>(input_idx)) {
+        auto expanded_handle = ExpandInputHelper<GPUBackend>(ws, input_idx, num_expand_dims);
+        AddInputHelper<GPUBackend>(expanded_handle);
+      } else {
+        auto expanded_handle = ExpandInputHelper<CPUBackend>(ws, input_idx, num_expand_dims);
+        AddInputHelper<CPUBackend>(expanded_handle);
+      }
+    }
```

Comment:
I insist on two things:
1. Verification that all expanded inputs (and arguments, for that matter) have the same number of frames in corresponding samples.
2. Expansion through broadcasting for frameless inputs.

Reply:
1. Added the validation.
2. Added a logic for handling the multi-inputs with broadcasting but left placeholder for the broadcasting itself: I'd rather do it as follow up if that's fine. It wont' be needed in the guassian blur anyway.